From 0dd80a34d0bbd7d330189105b3829c1ac6e63449 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 15 Sep 2017 10:48:19 +0800
Subject: [PATCH 2553/5242] MLK-16285-2 staging: typec: tcpci: add optional
 reset pin support

commit  dbb7fba24e89c89b8f4abc793f67a06bb217e4be from
https://source.codeaurora.org/external/imx/linux-imx.git

Some USB3 differential channel switch chips need to do reset before
functional, we add this support here.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Acked-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index c9bcdb5..e1a566c 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -737,6 +737,16 @@ static int tcpci_parse_config(struct tcpci *tcpci)
 static int tcpci_ss_mux_control_init(struct tcpci *tcpci)
 {
 	struct device *dev = tcpci->dev;
+	struct gpio_desc *gpiod_reset;
+
+	gpiod_reset = devm_gpiod_get_optional(dev, "reset", GPIOD_OUT_HIGH);
+	if (IS_ERR(gpiod_reset)) {
+		dev_err(dev, "Failed to request reset gpio.");
+		return PTR_ERR(gpiod_reset);
+	}
+
+	if (gpiod_reset)
+		usleep_range(700, 1000);
 
 	tcpci->ss_sel_gpio = devm_gpiod_get_optional(dev, "ss-sel",
 							GPIOD_OUT_HIGH);
-- 
1.7.9.5

